lsb_vsx: fix testcase 1 of fstat

adamdebek opened the pull request:
JIRA: CI-524

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes shortly -->
In the test, t1 is recorded after writing to the file. If the write occurs near the end of a second (e.g., 0.99s), the file’s st_mtime will be set to that second (0). Later, time(&t1) may return the next second (e.g., 1). The test then checks if st_mtime == t1 || t1+1 || t1+2, which can fail because st_mtime is actually earlier than t1. If t1 were collected before the write, the comparison would correctly account for the modification time, avoiding this race condition.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

<!--- In case of breaking change - please advice here what needs to be done in dependent projects. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here).

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

damianloew commented:
Could you briefly recall why it is needed?

damianloew reviewed (changes requested):
It looks like the changes are not only about fixing the described problem - please split to separate commits if needed and provide WHY in commit description.

adamdebek committed:
lsb_vsx: fix testcase 1 of fstat

Fix race condition and also uncomment a previously commented-out code section,
since we do not allow unnecessary manual tweaks.

JIRA: CI-524

adamdebek commented on lsb_vsx/patches/ps_VSX/16_fstat.patch:
```diff
@@ -1,22 +1,47 @@
++	(void) close(fd);
  
- 	DBUG_VOID_RETURN;
- }
```
The patch changed, I did not remove this myself.
